logstash-core/lib/logstash/java_pipeline.rb

```diff
@@ -65,6 +65,7 @@ def initialize(pipeline_config, namespaced_metric = nil, agent = nil)
     @flushing = java.util.concurrent.atomic.AtomicBoolean.new(false)
     @flushRequested = java.util.concurrent.atomic.AtomicBoolean.new(false)
     @shutdownRequested = java.util.concurrent.atomic.AtomicBoolean.new(false)
+    @crash_detected = Concurrent::AtomicBoolean.new(false)
```

Comment:
I had thought we could rely on the difference between `finished_run` and `finished_execution` to tell us whether we had crashed, because `finished_run` is only set in a non-exceptional path, but it turns out that the execeptions from a crash are already handled by the time it gets there.

Ideally we would have a finite state machine for the possible states of a given pipeline instead of having state sprinkled throughout, but that task was a much larger lift than simply adding a single flag.

I found a dirty way to abuse the ruby filter to cause an exception that escapes its handling and causes the pipeline to crash, emulating the type of crash that is possible by a plugin that fails to handle exceptions:

> ~~~ yaml
> - pipeline.id: uppy
>   config.string: |
>     input { heartbeat {} }
>     output { sink {} }
> - pipeline.id: crashy
>   config.string: |
>     input { heartbeat { } }
>     filter {
>       ruby {
>         init => "@poison = ::Class.new(::Exception) { def backtrace; throw(:boom); end; }"
>         code => "fail @poison.new"
>       }
>     }
>     output { sink {} }
>     output { stdout { codec => rubydebug } }
> ~~~